engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
@vrozov explicit override at application level isn't possible as the user does not know where in the cluster the AM will run. The only option the user is left with is to get the host configurations changed and that's a high bar. (I don't consider hacking the Apex code like I had to an option). It is better to do what we reasonably can to get a good out of the box behavior.